File: data_format_translation_servers/python/reader.py
import xml.etree.ElementTree as ET
import yaml
import csv
import json

def read_txt(path):
    with open(path) as f:
        text = f.read()
        return(text)

def read_xml(path):
    tree = ET.parse(path)
    root = tree.getroot()
    tagText = []
    for child in root:
        tagText.append([child.tag, child.text])
    return tagText

def read_yaml(path):
    with open(path) as f:
        data = yaml.safe_load(f)
        return data

def read_csv(path):
    data = []
    with open(path) as f:
        reader = csv.reader(f)
        for row in reader:
            data.append(row)
        f.close()
    return data

def read_json(path):
    with open(path) as f:
        data = json.load(f)
        return(data)

File: data_format_translation_servers/node/reader.js
import fs from 'fs';
import { xml2js } from 'xml-js';
import yaml from 'js-yaml';
import csv from 'csv-parser';

const readTxt = (path) => {
   try {
      const file = fs.readFileSync(path, 'utf-8');
      const text = file.toString();
      return text;
   } catch (error) {
      console.log(error);
   }
};

const readXml = (path) => {
   try {
      const file = fs.readFileSync(path, 'utf-8');
      const object = xml2js(file);
      return object;
   } catch (error) {
      console.log(error);
   }
};

const readYaml = (path) => {
   try {
      const file = fs.readFileSync(path, 'utf-8');
      const data = yaml.load(file);
      return data;
   } catch (error) {
      console.log(error);
   }
};

const readCsv = (path) => {
   return new Promise((resolve, reject) => {
      const results = [];
      const readStream = fs.createReadStream(path);

      readStream.pipe(csv()).on('data', (data) => results.push(data));

      readStream.on('error', () => {
         reject(e);
      });

      return readStream.on('end', () => {
         resolve(results);
      });
   });
};

const readJson = async (path) => {
   try {
      const file = fs.readFileSync(path);
      const data = JSON.parse(file);
      return data;
   } catch (error) {
      console.log(error);
   }
};

export { readTxt, readJson, readCsv, readXml, readYaml };
